express-api/src/services/keycloak/keycloakSchemas.ts

```diff
@@ -0,0 +1,37 @@
+import { array, z } from 'zod';
+
+// A role as returned from Keycloak.
+export const keycloakRoleSchema = z.object({
+  name: z.string(),
+  composite: z.boolean().optional(),
+});
+
+// The generic error object that Keycloak returns.
+export const keycloakErrorSchema = z.object({
+  message: z.string(),
+});
+
+// A list of roles returned by Keycloak.
+export const keycloakUserRolesSchema = z.object({
+  data: array(keycloakRoleSchema),
```

Comment:
```suggestion
  data: z.array(keycloakRoleSchema),
```
Then remove array import at top of file.